src/AstParser.js

```diff
@@ -207,7 +207,7 @@ class AstParser {
     let matches;
 
     // extract source fragments and inline expressions
-    while (matches = /@{(.*?)}/.exec(fragment)) {
+    while (matches = /@{([^@\r\n]*)}/.exec(fragment)) {
```

Comment:
Another problem with this change is that it prevents the code fragment from containing the `@` symbol.